blueprints/geometry/operations.py

```diff
@@ -0,0 +1,83 @@
+"""Geometry operations module."""
+
+import math
+from enum import Enum, auto
+
+from shapely import Point
+
+from blueprints.type_alias import RAD
+
+
+class CoordinateSystemOptions(Enum):
+    """Enum of the coordinate system options."""
+
+    XY = auto()
+    XZ = auto()
+    YZ = auto()
+
+
+def calculate_rotation_angle(
+    start_point: Point,
+    end_point: Point,
+    coordinate_system: CoordinateSystemOptions = CoordinateSystemOptions.XY,
+) -> RAD:
+    """
+    Calculates rotation of an end point in relation to the start point in a given plane/coordinate system [rad].
+
+    - Start point lies in the origin center of the quadrant.
+    - The rotation is calculated in the given plane in relation to the start point.
+    - The rotation is calculated in the range [0, 2*pi].
+    - The rotation is calculated in the counter-clockwise direction.
+
+    Parameters
+    ----------
+    start_point: Point
+        Starting point that will be used as reference.
+    end_point: Point
+        End point
+    coordinate_system: CoordinateSystemOptions
+        Desired plane that will be used as reference to calculate the rotation. Standard is XY-plane.
+
+    Returns
+    -------
+    RAD
+        rotation of end point relative to begin node in radians in the given plane [rad].
+
+    Raises
+    ------
+    ValueError
+        If start_point and end_point are the same.
+        If start_point or end_point do not have z value when rotation angle in XZ or YZ plane is requested.
+    """
+    if list(start_point.coords) == list(end_point.coords):
```

Comment:
Have you considered using the euqals_exact() method from Shapely?

https://shapely.readthedocs.io/en/stable/reference/shapely.Point.html#shapely.Point.equals_exact